File: services/auth-service/src/main/java/com/mealmap/authservice/core/dto/UserDto.java
package com.mealmap.authservice.core.dto;

import com.mealmap.authservice.core.enums.Role;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.util.UUID;

@Data
@Builder
@Schema(description = "Информация о пользователе")
public class UserDto {
    private UUID id;

    private String phoneNumber;

    private String email;

    private String firstName;

    private String lastName;

    private Integer organizationId;

    private UserStatus status;

    private Role role;

    private LocalDate createdAt;
}

File: frontend/src/api/auth/dto/UserDto.ts
import { Role } from "../enums/Role";
import { UserStatus } from "./UserStatus";

export interface UserDto {
  id: string; // UUID
  phoneNumber: string;
  email: string;
  firstName: string;
  lastName: string;
  organizationId: number;
  status: UserStatus;
  role: Role;
  createdAt: string; // ISO date string
}
